cloudevents/client.go

```diff
@@ -0,0 +1,131 @@
+package cloudevents
+
+import (
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+)
+
+// Client wraps Builder, and is intended to be configured for a single event
+// type and target
+type Client struct {
+	Builder
+	Target string
+
+	send chan interface{}
+	done chan bool
+}
+
+const (
+	// MAX_SEND_CHANNEL is the max buffered channel length for the send channel
+	MAX_SEND_CHANNEL = 10
+)
+
+// NewClient returns a CloudEvent Client used to send CloudEvents. It is
+// intended that a user would create a new client for each tuple of eventType
+// and target. This is a n optional helper method to avoid the tricky creation
+// of the embedded Builder struct.
+func NewClient(eventType, source, target string) *Client {
+	c := &Client{
+		Builder: Builder{
+			Source:    source,
+			EventType: eventType,
+		},
+		Target: target,
+	}
+	return c
+}
+
+// Send creates a request based on the client's settings and sends the data
+// struct to the target set for this client. It returns error if there was an
+// issue sending the event, otherwise nil means the event was accepted.
+func (c *Client) Send(data interface{}) error {
+	resp, err := c.RequestSend(data)
+	if err != nil {
+		return err
+	}
+	if accepted(resp) {
+		return nil
+	}
+	return fmt.Errorf("error sending cloudevent: %s", status(resp))
+}
+
+// RequestSend uses the internal builder to make a request with the provided
+// data struct using the previously set parameters of event type and target.
+// Use this instead of client.Send() if processing of the http response directly
+// is required.
+func (c *Client) RequestSend(data interface{}) (*http.Response, error) {
```

Comment:
Why are both `Send` and `RequestSend` exposed publicly? (Event more strangely, they seem to only differ in return type, where `RequestSend` also returns an `http.Response`.

Can we change the `Send` contract to return `(*http.Response, error)`, and return both the Response and a non-nil error when the response is not `accepted()`? Users who aren't interested in the Response can simply write:

```golang
if _, err := c.Send(myStuff); err != nil {
  return err
}
```

Reply:
This leaves the Response open though right? My goal was to close the http request and read the body out for the user of the lib. So I did not want to return the `http.Response` back if that was the case. Plus I wrote some logic to understand the response and did not want to expose that to the client user. I just want them to know if it worked out not.